pkg/splunk/enterprise/clustermaster.go

```diff
@@ -137,6 +137,14 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterpriseApi.Clu
 	if err != nil {
 		return result, err
 	}
+
+	//make changes to respective mc configmap when changing/removing mcRef from spec
+	extraEnv, err := VerifyCMisMultisite(cr, namespaceScopedSecret)
+	err = validateMonitoringConsoleRef(client, statefulSet, extraEnv)
+	if err != nil {
+		return result, err
+	}
+
```

Comment:
I think, moving the `validateMonitoringConsoleRef`(and also renaming it as `validateAndMonitoringConsoleRef`, then, also covering the case where the MC ref is added for the very first time) to the `ready phase checking` will reduce the redundant calling across the reconciliations.

Reply:
Discussed with @sgontla offline. When adding mcRef for the first time we need some information from the CM, and that should be in the ready phase, so we still need to perform that in that ready phase. For re-config, we need to update MC configmap before update() kicks in. So separating both the cases